src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+---
+layout: shared/narrow
+published: false
+title: "Setting up a basic service worker"
+description: "This section describes the most basic service worker possible. It shows you how a client gets, or registers, a service worker. It shows you how a service worker prepares to act as a proxy for its clients. Both subjects have more depth than is shown in this section. It's not intended to be comprehensive; It's intended to give you a basic foundation on which to build service worker knowledge."
+authors:
+  - josephmedley
+published_on: 2015-10-01
+updated_on: 2015-10-01
+order: 4
+key-takeaways:
+  tldr:   
+  - "Call <code>register()</code> to install a service worker." 
+  - "Learn to use promises before attempting service workers."
+  - "Service workers have a limited scope and requrie HTTPS."
+  - "The install process has two events: install and activate."
+  - "The user has to navigate before the service worker start proxying."
+  - "Call <code>waintUntil()</code> to keep a service worker out of its own way."
+notes:
+  promises:
+    - "<b>Promises</b>&mdash;Notice the use of <code>.then()</code> at the end of the <code>register()</code> function. This is an example of an ECMAScript 2015 construct called a <a href='http://www.html5rocks.com/en/tutorials/es6/promises/'>promise</a>. Service workers make heavy use of promises. If you've never used promises before, you should familiarize yourself with them before trying to implement a service worker."
+  https-only:
+    - "<b>HTTPS Only</b>&mdash;Service workers can do almost whatever they want to HTTP requests and responses. Since this would make them targets for man-in-the-middle attacks, they must be served over HTTPS. This doesn't mean you need HTTPS for development and testing. Service workers served over localhost will also work."
+  sws-dont-control:
+    - "<b>Service Workers Don't Take Control</b>&mdash;If you look around the web, you'll find that many of the pages discussing service workers refer to the service worker as 'taking control' of a page. But as we saw earlier, a service workers can't change a page's DOM. That's why it may be more accurate to think of a service worker as ready to proxy."
```

Comment:
Could be misleading as `navigator.serviceWorker.controller` is a service worker. Instead of "Service Workers Don't Take Control", something like "Isolated from pages", or "Independent to pages"?